### PR TITLE
Allow ordering media by title

### PR DESCRIPTION
**Pull request by mfairchild365**
This allows ordering media by title. It also re-works the filter and sort nav so that it is re-used and more accessible.

### CODE REVIEW

**Comment by mfairchild365** [comment]
@spennythug I assigned to you for visual feedback. @kabel is welcome to provide feedback too. :)

File: www/templates/html/MediaList.tpl.php [hunk 1]
```diff
@@ -61,17 +61,52 @@
...
+
+        <?php
+        $buttons = new stdClass();
+        $buttons->selected_key = $context->options['orderby'];
+        $buttons->group_id = 'orderby';
+        $buttons->buttons = [
+            'datecreated' => [
+                'label' => 'Recent',
+                'url' => $context->getURL(array('orderby' => 'datecreated', 'order' => 'DESC'))
+            ],
```
Comment (spennythug) [inline www/templates/html/MediaList.tpl.php]: UGGGGGHHHHH!! WHITESPACE! *barfs*

Comment (mfairchild365) [inline www/templates/html/MediaList.tpl.php]: What whitespace?

Comment (spennythug) [inline www/templates/html/MediaList.tpl.php]: GET RID OF IT

File: www/templates/html/FeedList.tpl.php [hunk 2]
```diff
@@ -47,13 +47,26 @@
...
     <div class="wdn-inner-wrapper wdn-inner-padding-none">
-        <p class="mh-sort-filter">
-            <span class="mh-btn-group">
-                <a href="/channels/" class="wdn-button wdn-button-brand <?php echo (($context->options['orderby'] != 'datecreated') && ($context->options['orderby'] != 'plays')) ? ' active' : '' ?>">Alphabetical</a>
-                <a href="?orderby=datecreated&amp;order=DESC" class="wdn-button wdn-button-brand <?php echo ($context->options['orderby'] == 'datecreated') ? ' active' : '' ?>">Recent</a>
-                <a href="?orderby=plays&amp;order=DESC" class="wdn-button wdn-button-brand <?php echo ($context->options['orderby'] == 'plays') ? ' active' : '' ?>">Popular</a>
-            </span>
-        </p>
+        <?php
+        $buttons = new stdClass();
```
Comment (kabel) [inline www/templates/html/FeedList.tpl.php]: As a best practice, I would _not_ instantiate a `stdClass`. I consider it an internal PHP class that happens to be usable from userland. Instead, I would use an array and _cast_ it using `(object) $buttons`.

**Comment by kabel** [comment]
Hey! We can claim this as a feature that YouTube doesn't have 😉

File: www/templates/html/mh-sort-filter.tpl.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <span id="mh-sort-filter-<?php echo $context->group_id; ?>" class="sort-filter-title">Order By:</span>
+    <ul class="mh-btn-group">
+
+        <?php foreach ($context->buttons as $key=>$details):?>
+            <?php
+            $is_current_option = ($context->selected_key == $key) ? true : false;
+            $active_class = $is_current_option ? 'active' : '';
+            ?>
+            <li>
+                <a href="<?php echo htmlentities($details['url'], ENT_QUOTES) ?>" class="wdn-button wdn-button-brand <?php echo $active_class ?>">
```
Comment (kabel) [inline www/templates/html/mh-sort-filter.tpl.php]: Does Savvy not escape this be default? If not, perhaps `$savvy->escape(...)` would keep it consistent.

File: www/templates/html/MediaList.tpl.php [hunk 4]
```diff
@@ -61,17 +61,52 @@
...
-            </span>
-            <span class="mh-btn-group">
-                <a href="<?php echo htmlentities($context->getURL(array('f' => '')), ENT_QUOTES)  ?>" class="wdn-button<?php echo ($context->options['f'] == '') ? ' active' : '' ?>">All</a>
-                <a href="<?php echo htmlentities($context->getURL(array('f' => 'audio')), ENT_QUOTES) ?>" class="wdn-button<?php echo ($context->options['f'] == 'audio') ? ' active' : '' ?>">Audio</a>
-                <a href="<?php echo htmlentities($context->getURL(array('f' => 'video')), ENT_QUOTES) ?>" class="wdn-button<?php echo ($context->options['f'] == 'video') ? ' active' : '' ?>">Video</a>
-            </span>
-        </p>
+
+        <?php
+        $buttons = new stdClass();
```
Comment (kabel) [inline www/templates/html/MediaList.tpl.php]: Same thing about best practice with stdClass.

File: www/templates/html/FeedList.tpl.php (hunk 2 above)
Comment (mfairchild365) [inline www/templates/html/FeedList.tpl.php]: I'm going to do `$buttons = (object)[];` and you can't stop me.

Comment (kabel) [inline www/templates/html/FeedList.tpl.php]: Lol, why mix object and array syntax? You must just like the `->` operator

Comment (spennythug) [inline www/templates/html/FeedList.tpl.php]: FIGHT THE POWER!

Comment (mfairchild365) [inline www/templates/html/FeedList.tpl.php]: I'm all about the `->` operator

File: www/templates/html/mh-sort-filter.tpl.php (hunk 3 above)
Comment (mfairchild365) [inline www/templates/html/mh-sort-filter.tpl.php]: You are not wrong, and I would like to see this fixed. However I think it is out of scope of the current PR. Savvy in the project has never been configured to escape... and I'm worried about double encoding if I turn it own now without being super careful.

Comment (kabel) [inline www/templates/html/mh-sort-filter.tpl.php]: Eek. We could be rife with injections!

**Pull request merged**